PILOT-1718: Update unit tests for kafka client

afmam opened the pull request:
## Summary

- Improved the kafka client unit tests by:
      1. Testing business logic separately (message creation/validation) 
      2. Mocking broker connection and message sending. Tests will not fail non-deterministically, as a broker is not required.
- Updated kafka client to separate logic for querying attribute templates.

## JIRA Issues

https://indocconsortium.atlassian.net/browse/PILOT-1718

## Type of Change

- [X] Refactor or reformatting

## Testing

Are there any new or updated tests to validate the changes?

- [X] Yes
- [ ] No

## Test Directions

- Tests should not fail non-deterministically when merged to `dev` branch.

erikdvlp reviewed (changes requested):
I am still seeing inconsistent unit test results locally; [see logs](https://github.com/PilotDataPlatform/metadata/files/9830512/unit_tests.txt). Lines 69-275 contain a failed run.

afmam commented:
> I am still seeing inconsistent unit test results locally; [see logs](https://github.com/PilotDataPlatform/metadata/files/9830512/unit_tests.txt). Lines 69-275 contain a failed run.

@erikdvlp The issue seemed to be the `generate_random_{}` function, which periodically introduced whitespace into the names. With `container_code` for example, querying for an item by a `container_code` with a whitespace would result in not finding the item, and therefore test(s) failing: https://github.com/PilotDataPlatform/metadata/pull/66/files#diff-e52e4ddd58b7ef887ab03c04116e676f6280b824ab7469d5d3080e5cba4f2128R33

I also noticed the following bugs in the tests that I resolved as well, which were:

1. During teardown of `test_items` fixture, each item is to be deleted, however, once the folder is deleted, all of its children created are also deleted (file_1, file_2, file_3) automatically. When trying to delete these 3 files after the folder, they are not found and raise an error. Therefore, only the name_folder and folder need to be called for deletion: https://github.com/PilotDataPlatform/metadata/pull/66/files#diff-e52e4ddd58b7ef887ab03c04116e676f6280b824ab7469d5d3080e5cba4f2128R148
2.  During teardown of `test_favourites`, deletion of the collection would fail, as the logic for deleting favourites would assume an entity is of type 'item' by default. I added **type** to the params to avoid this: https://github.com/PilotDataPlatform/metadata/pull/66/files#diff-e52e4ddd58b7ef887ab03c04116e676f6280b824ab7469d5d3080e5cba4f2128R225
3. For deleting an item, `combine_item_tables()` was called on NoneType obj if the entity did not exist. I changed this by validating that the entity exists first: https://github.com/PilotDataPlatform/metadata/pull/66/files#diff-e65a226f96a2ab40e6dbc4631fbbb066ec33a48d15c2b13b0ff3f1b393c1928bR482

erikdvlp commented:
> > I am still seeing inconsistent unit test results locally; [see logs](https://github.com/PilotDataPlatform/metadata/files/9830512/unit_tests.txt). Lines 69-275 contain a failed run.
> 
> @erikdvlp The issue seemed to be the `generate_random_{}` function, which periodically introduced whitespace into the names...

Wow, great work. Thanks for catching all of this. I'm not seeing any failed runs locally.
